cmd/restore.go

```diff
@@ -0,0 +1,269 @@
+
+	entries, err := os.ReadDir(srcDir)
+	if err != nil {
+		return nil, util.Error(err, "error reading directory")
+	}
+
+	// See if collections need to be filtered
+	collPattern := `.*`
+	if len(collectionFilter) > 0 {
+		collPattern = strings.Join(collectionFilter, "|")
+	}
+
+	// Obtain schema names from json files
+	backupPattern := fmt.Sprintf(
+		`^%s[.](?P<collection>(%s))[.]%s$`,
+		db,
+		collPattern,
+		backupFileExtension)
+	re := regexp.MustCompile(backupPattern)
+
+	for _, v := range entries {
+		if v.IsDir() {
+			continue
+		}
+
+		// Extract database name from filename
+		match := re.FindStringSubmatch(v.Name())
+		if len(match) == 0 {
+			continue
+		}
+
+		collections = append(collections, match[1])
+	}
+
+	return collections, nil
+}
+
+func createDatabase(ctx context.Context, db string) {
+	withLogin(ctx, func(ctx context.Context) error {
+		err := client.Get().CreateDatabase(ctx, db)
+		return util.Error(err, "create database")
+	})
+}
+
+func restoreDatabase(ctx context.Context, db, path string) error {
+	docs := make([]json.RawMessage, 0)
+
+	f, err := os.Open(path)
+	if err != nil {
+		return util.Error(err, "failed to read schema")
+	}
+	defer f.Close()
+
+	dec := json.NewDecoder(f)
+	for dec.More() {
+		var v json.RawMessage
+		err := dec.Decode(&v)
+		util.Fatal(err, "reading documents from stream of documents")
+
+		docs = append(docs, v)
+	}
+
+	restoreDB := amendDatabaseName(db)
+	createDatabase(ctx, restoreDB)
+
+	return client.Transact(ctx, restoreDB, func(ctx context.Context, tx driver.Tx) error {
+		for _, v := range docs {
+			if err := createCollection(ctx, tx, driver.Schema(v)); err != nil {
+				return util.Error(err, "failed to create collection")
+			}
+		}
+
+		return nil
+	})
+}
+
+func restoreCollection(ctx context.Context, db, collection, path string) error {
+	docs := make([]json.RawMessage, 0)
+
+	f, err := os.Open(path)
+	if err != nil {
+		return util.Error(err, "failed to read collection")
+	}
+	defer f.Close()
+
+	dec := json.NewDecoder(f)
+	for dec.More() {
+		var v json.RawMessage
+		err := dec.Decode(&v)
+		util.Fatal(err, "reading documents from stream of documents")
+
+		docs = append(docs, v)
+	}
+
+	if len(docs) == 0 {
+		return nil
+	}
+
+	restoreDB := amendDatabaseName(db)
+
+	return client.Transact(ctx, restoreDB, func(ctx context.Context, tx driver.Tx) error {
+		ptr := unsafe.Pointer(&docs)
+		_, err := client.Get().UseDatabase(restoreDB).Insert(ctx, collection, *(*[]driver.Document)(ptr))
+		if err != nil {
+			return util.Error(err, "failed to insert doc")
+		}
+
+		return nil
+	})
+}
+
+var restoreCmd = &cobra.Command{
+	Use:   "restore [filters]",
+	Short: "restores documents and schemas from JSON files",
+	Long: `restores documents and schemas from JSON files in a directory specified in the argument.
+
+	If a database filter is provided it only restores the schemas of the databases specified.
+	Likewise, collection filters will limit the input to matching collection names.`,
```

Comment:
It would really useful if we have example commands here.

Reply:
Added, thanks for the feedback!